http-server/src/main/java/org/triplea/server/http/ServerApplication.java

```diff
@@ -59,50 +62,50 @@ public void initialize(final Bootstrap<AppConfig> bootstrap) {
 
   @Override
   public void run(final AppConfig configuration, final Environment environment) {
-    environment.jersey().register(new ClientExceptionMapper());
-
     if (configuration.isProd()) {
       configuration.verifyProdEnvironmentVariables();
     }
 
-    final JdbiFactory factory = new JdbiFactory();
-    final Jdbi jdbi = factory.build(environment, configuration.getDatabase(), "postgresql-connection-pool");
+    if (configuration.isLogRequestAndResponses()) {
+      environment.jersey().register(
+          new LoggingFeature(Logger.getLogger(LoggingFeature.DEFAULT_LOGGER_NAME),
+              Level.INFO,
+              LoggingFeature.Verbosity.PAYLOAD_ANY, LoggingFeature.DEFAULT_MAX_ENTITY_SIZE));
+    }
 
-    // register all endpoint handlers here:
-    environment.jersey().register(errorReportController(configuration, jdbi));
+    exceptionMappers()
+        .forEach(mapper -> environment.jersey().register(mapper));
+
+    endPointControllers(configuration, environment)
+        .forEach(controller -> environment.jersey().register(controller));
   }
 
-  private static ErrorReportController errorReportController(
```

Comment:
This code is moved to ErrorReportControllerFactory. It follows a pattern where there is a "*Factory" class that creates each controller, hence we do not have to construct each controller in the server application class.